zmon-controller-app/src/main/java/org/zalando/zmon/api/AlertStatusAPI.java

```diff
@@ -135,4 +157,45 @@ public JsonNode getAlertStatus(@PathVariable("ids") final List<String> ids) thro
 
         return new ResponseEntity<>(alerts, HttpStatus.OK);
     }
+
+    @RequestMapping(value = "/alert-results", method = RequestMethod.GET)
+    public ResponseEntity getAlertResults(
+        @RequestParam(value = "filter") final String filters
+    ) {
+        final JsonNode filter = parseFilter(filters);
+        if (!isAllowed(filter)) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        final ArrayNode filterArray = mapper.createArrayNode();
+        filterArray.add(filter);
+
+        return new ResponseEntity<>(new AlertResults(service.getAlertResults(filterArray)), HttpStatus.OK);
+    }
+
+    private boolean isAllowed(JsonNode filter) {
+        if (filter == null || !filter.isObject() || filter.size() == 0) return false;
+
+        final Iterator<String> keys = filter.fieldNames();
+        boolean hasAtLeastOneFilterSet = false;
+        while (keys.hasNext()) {
```

Comment:
any particular reason for `while ... hasNext` instead of `foreach` loop?

Reply:
`JsonNode#fieldNames()` returns `Iterator<String>` (not an `Iterable<String>`).